### PR TITLE
add SPIR-V 1.4 testing for various miscellaneous additions

### PR DESCRIPTION
**Pull request by bashbaug**
This PR adds targeted testing for a few remaining miscellaneous SPIR-V 1.4 features:

* Variables in the Function storage class can have a NonWriteable decoration.
* OpCopyMemorySized can have two optional memory operands.
* OpSelect can have composite operands that are not vectors.

### CODE REVIEW

**Comment by bashbaug** [comment]
I fixed the merge conflict in this PR and in #2053.  I don't have a strong opinion which goes in first.  If it's not this one though, I'll probably need to pull in the latest changes one more time.

File: test_conformance/spirv_new/test_spirv_14.cpp [hunk 1]
```diff
@@ -352,3 +352,162 @@ TEST_SPIRV_FUNC(spirv14_usersemantic_memberdecoratestring)
...
+                                results.size() * sizeof(results[0]),
+                                results.data(), 0, nullptr, nullptr);
+    SPIRV_CHECK_ERROR(error, "Unable to read destination buffer");
+
+    const TestStruct expected_even{ 1024, 3.1415f };
+    const TestStruct expected_odd{ 2048, 2.7128f };
+
+    for (size_t i = 0; i < global_size; i++)
+    {
+        const TestStruct expected = (i & 1) ? expected_even : expected_odd;
```
Comment (svenvh) [inline test_conformance/spirv_new/test_spirv_14.cpp:502]:
```suggestion
        const TestStruct &expected = (i & 1) ? expected_even : expected_odd;
```

File: test_conformance/spirv_new/test_spirv_14.cpp [hunk 2]
```diff
@@ -352,3 +352,162 @@ TEST_SPIRV_FUNC(spirv14_usersemantic_memberdecoratestring)
...
+                                   nullptr, 0, nullptr, nullptr);
+    SPIRV_CHECK_ERROR(error, "Failed to enqueue kernel");
+
+    error = clEnqueueReadBuffer(queue, dst, CL_TRUE, 0,
+                                results.size() * sizeof(results[0]),
+                                results.data(), 0, nullptr, nullptr);
+    SPIRV_CHECK_ERROR(error, "Unable to read destination buffer");
+
+    const TestStruct expected_even{ 1024, 3.1415f };
+    const TestStruct expected_odd{ 2048, 2.7128f };
```
Comment (svenvh) [inline test_conformance/spirv_new/test_spirv_14.cpp:498]: I think "even" and "odd" are mixed up here; perhaps just rename them to "struct_a", "struct_b" to align with the names in the SPIR-V assembly.

File: test_conformance/spirv_new/test_spirv_14.cpp (hunk 1 above)
Comment (bashbaug) [reply test_conformance/spirv_new/test_spirv_14.cpp]: fixed!

File: test_conformance/spirv_new/test_spirv_14.cpp (hunk 2 above)
Comment (bashbaug) [reply test_conformance/spirv_new/test_spirv_14.cpp]: Good idea - I switched the names to "struct_a" and "struct_b" to match the SPIR-V source.

**Review by svenvh — APPROVED** [review]

**Comment by bashbaug** [comment]
Merging as per agreement in the October 29th teleconference.

**Pull request merged**